fix: prevent data/configuration mismatch [RWDQA-60]

tomzemp opened the pull request:
See https://dhis2.atlassian.net/browse/RWDQA-60

This PR
- Sections 2,3,4: updates custom hooks to set parameters used for fetching data, this allows the calculations to be performed with the configurations/parameters that correspond with the data and prevents the errors caused by mismatch between the two
- Section 1: refactors to use a custom hook and then makes the above change. The refactoring into a custom hook will also be of use when we filter out the indicators from this section (https://dhis2.atlassian.net/browse/RWDQA-65) as we will then need to check which items are data elements/indicators.

KaiVandivier commented:
Nice, tested it and it works 🙂

What do you think about a solution like [this one that I drafted](https://github.com/hisprwanda/who-data-quality-annual-report/commit/a136e1110ef00898dbf4445d94413f1b20dc1093)? To me, it feels like addresses the problem closer to its root, which saves some new objects and optional chaining

tomzemp commented:
> What do you think about a solution like [this one that I drafted](https://github.com/hisprwanda/who-data-quality-annual-report/commit/a136e1110ef00898dbf4445d94413f1b20dc1093)? To me, it feels like addresses the problem closer to its root, which saves some new objects and optional chaining

Nice. I like the idea of doing the calculations as a direct response to the data fetching. I can take that idea and apply to the other sections and commit

tomzemp commented:
@KaiVandivier : what if we move the calculations for the section into the custom hook (section 2 for [example](https://github.com/hisprwanda/who-data-quality-annual-report/commit/5dae96c2d7961efa4367cb1a8ba126ce8b49cfb4))? Thereby we perform the calculations as a direct response to the data fetching, we reuse the parameters that we're already passing to the refetch, and we keep the custom hook a bit closer to our useDataQuery pattern (we request some data, and then we display it when the hook returns it).

The example approach here will also have the "advantage" of wrapping the calculations in the try/catch statement of the hook (so the app won't be crashed if something goes wrong in the calculations, though this is admittedly not really best practice to just move to ever less-specific error catching 😅)

KaiVandivier commented:
@tomzemp Yeah! That seems great 🙂 want to call the resulting hook `useSectionTwoData()` then, since it does both fetching and calculating?

tomzemp commented:
> want to call the resulting hook useSectionTwoData() then, since it does both fetching and calculating?
@KaiVandivier: Sounds good. I've made that update and have applied similar logic to the other section2

I've also moved the names used for the responses (e.g. `OVERALL_RESPONSE`) from the hooks to the calculations for sections 3 and 4 (to avoid having calculations import from hook, and hook import from calculations) and moved them into a separate file for section 2.

KaiVandivier reviewed (approved):
Nice, looks good 👍
